src/sentry/rules/conditions/event_frequency.py

```diff
@@ -125,3 +127,75 @@ def query_hook(self, event, start, end, environment_id):
             environment_id=environment_id,
             use_cache=True,
         )[event.group_id]
+
+
+session_percent_intervals = {
+    "1m": ("one minute", timedelta(minutes=1)),
+    "5m": ("five minutes", timedelta(minutes=5)),
+    "10m": ("ten minutes", timedelta(minutes=10)),
+    "30m": ("30 minutes", timedelta(minutes=30)),
+    "1h": ("one hour", timedelta(minutes=60)),
+}
+
+
+class SessionPercentForm(EventFrequencyForm):
+    intervals = session_percent_intervals
+    interval = forms.ChoiceField(
+        choices=[
+            (key, label)
+            for key, (label, duration) in sorted(
+                session_percent_intervals.items(),
+                key=lambda key____label__duration: key____label__duration[1][1],
+            )
+        ]
+    )
+    value = forms.IntegerField(widget=forms.TextInput())
+
+
+class SessionPercentCondition(BaseEventFrequencyCondition):
+    label = "The issue has more errors than {value} percent of sessions in {interval}"
+
+    def __init__(self, *args, **kwargs):
+        self.intervals = session_percent_intervals
+        self.form_cls = SessionPercentForm
+        super().__init__(*args, **kwargs)
+
+    def query_hook(self, event, start, end, environment_id):
+        cache_key = f"r.c.spc:{environment_id}-{start.replace(minute=0, second=0, microsecond=0, tzinfo=None)}-{end.replace(minute=0, second=0, microsecond=0, tzinfo=None)}".replace(
+            " ", ""
+        )
+        session_count = cache.get(cache_key)
+        if session_count is None:
+            filters = {"project_id": [event.project_id]}
+            if environment_id:
+                filters["environment"] = [environment_id]
+
+            start = start - timedelta(days=5)
+            end = end + timedelta(days=5)
```

Comment:
Why do we fetch 5 days on each side?

Reply:
Oops, I added this in when testing something and never removed it :|